### PR TITLE
Upgraded PPT for more variety of MR creation options. Added tests

### PR DESCRIPTION
**Pull request by bcornwellmott**
The Production Planning Tool now has the following options:
1. Export all raw materials (same as before)
2. Export one level of materials (same as before)
3. Export all levels of materials (including sub-contracted raw materials)
4. Export all levels of materials except sub-contracted raw materials)
5. Export one level materials (except manufactured items)
6. Export only raw materials (except sub-contracted raw materials)

The above six outcomes are tested with the test_production_planning_tool.py code.

### CODE REVIEW

**Comment by nabinhait** [comment]
screen shot 2016-08-17 at 12 01 09 pm

File: erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py [hunk 1]
```diff
@@ -326,28 +326,40 @@ def get_raw_materials(self, bom_dict):
...
 				for so_qty in so_wise_qty:
 					item_list.append([item, flt(item_details.qty) * so_qty[1], item_details.description,
 						item_details.stock_uom, item_details.min_order_qty, so_qty[0]])
-
+						
 		self.make_items_dict(item_list)
 
+	def get_subitems(self,bom_wise_item_details, bom, parent_qty, include_sublevel, only_raw, supply_subs):
+		for d in frappe.db.sql("""select bom_item.item_code, default_material_request_type,
+						ifnull(%(parent_qty)s * sum(bom_item.qty/ifnull(bom.quantity, 1)), 0) as qty,
```
Comment (nabinhait) [inline erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py]: Use less indent

File: erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py [hunk 2]
```diff
@@ -326,28 +326,40 @@ def get_raw_materials(self, bom_dict):
...
+	def get_subitems(self,bom_wise_item_details, bom, parent_qty, include_sublevel, only_raw, supply_subs):
+		for d in frappe.db.sql("""select bom_item.item_code, default_material_request_type,
+						ifnull(%(parent_qty)s * sum(bom_item.qty/ifnull(bom.quantity, 1)), 0) as qty, 
+						item.is_sub_contracted_item as is_sub_contracted, item.default_bom as default_bom
+						from `tabBOM Item` bom_item, `tabBOM` bom, tabItem item
+						where bom.name = bom_item.parent and bom.name = %(bom)s and bom_item.docstatus < 2
+						and bom_item.item_code = item.name
+						and item.is_stock_item = 1
+						group by bom_item.item_code""", {"bom": bom, "parent_qty": parent_qty}, as_dict=1):
+							if (d.default_material_request_type == "Purchase" and not (d.is_sub_contracted and only_raw and include_sublevel)) or (d.default_material_request_type == "Manufacture" and not only_raw):
```
Comment (nabinhait) [inline erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py]: For better readability use multiline condition, use braces or backslash for multiline. Max chars in one line should be less than 99.

**Comment by bcornwellmott** [comment]
Updated.

**Comment by bcornwellmott** [comment]
@nabinhait, the fixes you requested have been implemented.

**Pull request merged**